### ISSUE TITLE
init fails on views

### ISSUE THREAD

**Issue by tomkrouper** [skeema/skeema#2]
I was attempting to test skeema, and I ran into an issue immediately on the import. I removed a view from the schema failing and it imported without an issue. However, the view is still part of the schema, so it needs to be there. Also the error message didn't tell me where or why it failed, so it was hard to just know that the issue was the VIEW from the beginning.

```
$ bin/skeema init -h <myhostname> --debug
2017-02-06 09:19:29 [INFO]  Creating and using host dir /Users/tomkrouper/test/<myhostname> for <myhostname>:3306

2017-02-06 09:19:29 [INFO]  Populating /Users/tomkrouper/test/<myhostname>/<first_schema_on_host>
2017-02-06 09:19:29 [ERROR] sql: Scan error on column index 9: unsupported Scan, storing driver.Value type <nil> into type *string
2017-02-06 09:19:29 [DEBUG] Exit code 2
```

**Comment by evanelias** [comment]
Thank you for the report! A fatal error upon encountering a view is definitely unexpected. Currently the [intended behavior](https://github.com/skeema/skeema/blob/master/doc/requirements.md#ignored-by-skeema) with views is to simply [ignore them](https://github.com/skeema/tengo/blob/master/schema.go#L87), so there must be a Skeema bug somewhere.

I'll try to repro the error. What version and flavor of MySQL are you using? Also, anything else unusual about that schema or the view?

It's possible the problem isn't necessarily just with the presence of a view; there might be some additional condition required to trigger it. I believe that error indicates that a nullable column is being read into a non-nullable Golang value, which may suggest some previously non-nullable `information_schema` column is nullable in a specific version or flavor/fork of MySQL.

**Comment by tomkrouper** [comment]
Once you see this view, I think the version won't matter, but it happened both on `Percona 5.6.32` and `MySQL 5.7.17 (Homebrew)`. This view is indeed offering up `NULL` on non-nullable fields in the second half of the `UNION`. So that is probably it.

```
mysql> show create table heartbeat_view\G
*************************** 1. row ***************************
                View: heartbeat_view
         Create View: CREATE ALGORITHM=UNDEFINED DEFINER=`root`@`localhost` SQL SECURITY INVOKER VIEW `heartbeat_view` AS select `heartbeat`.`server_id` AS `originating_server_id`,`heartbeat`.`ts` AS `master_ts`,now() AS `time_now`,timestampdiff(SECOND,`heartbeat`.`ts`,now()) AS `absolute_lag` from `heartbeat` union all select NULL AS `NULL`,NULL AS `NULL`,now() AS `now()`,NULL AS `NULL` order by `master_ts` desc limit 1
character_set_client: utf8
collation_connection: utf8_general_ci
1 row in set (0.00 sec)

mysql> show create table heartbeat\G
*************************** 1. row ***************************
       Table: heartbeat
Create Table: CREATE TABLE `heartbeat` (
  `ts` varchar(26) NOT NULL,
  `server_id` int(10) unsigned NOT NULL,
  `file` varchar(255) DEFAULT NULL,
  `position` bigint(20) unsigned DEFAULT NULL,
  `relay_master_log_file` varchar(255) DEFAULT NULL,
  `exec_master_log_pos` bigint(20) unsigned DEFAULT NULL,
  PRIMARY KEY (`server_id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8
1 row in set (0.00 sec)
```

**Comment by tomkrouper** [comment]
Though, I guess it really into `null` on a non-nullable table, since it doesn't reference a table in the second half of the query. None the less, likely an odd pattern.

**Comment by evanelias** [comment]
Ah my mistake, this is actually a bug that affects all views. I just hadn't tested the view-ignoring code in a few months since I've mostly been ignoring views for the initial alpha/beta/pre-release stage. (Although Skeema has *unit* tests, it doesn't yet have automated *integration* tests... I'm slowly working on a solution there, but it's complicated because the test setup needs to span multiple MySQL versions and flavors)

I'll comment on here once the fix is pushed to master, hopefully later today. Thanks again for the thorough report, much appreciated!

**Comment by evanelias** [comment]
This is now fixed in the master branch.

Looks like the problem was introduced about a month ago when I fleshed out the support for character sets and collations. Views don't have a table-level collation, so an information_schema column was coming back as NULL in a way not properly handled by Skeema. This query (as well as a few other information_schema queries) now gracefully handles NULL values in all relevant nullable information_schema cols.

I also improved the error messages in that method a bit, but unfortunately it's still tricky to figure out which table/view/column/etc is causing a problem. Skeema queries information_schema in bulk for each database's objects, rather than querying one table/view/etc at a time. But hopefully this method won't be the source of additional errors now that it is resilient to NULLs.
